doc/error_reporting.md

```diff
@@ -0,0 +1,88 @@
+```ruby
+Yast.import "Report"
+Yast::Report.Error("Something was wrong")
+```
+
+Usually, the reporting settings are defined in the AutoYaST profile. For instance, with the
+following settings, AutoYaST stops only when an error message is reported. For the rest, it applies
+a 10 seconds time-out. Additionally, all messages are logged.
+
+```xml
+<report>
+  <errors>
+    <show config:type="boolean">true</show>
+    <timeout config:type="integer">0</timeout>
+    <log config:type="boolean">true</log>
+  </errors>
+  <warnings>
+    <show config:type="boolean">true</show>
+    <timeout config:type="integer">10</timeout>
+    <log config:type="boolean">true</log>
+  </warnings>
+  <messages>
+    <show config:type="boolean">true</show>
+    <timeout config:type="integer">10</timeout>
+    <log config:type="boolean">true</log>
+  </messages>
+  <yesno_messages>
+    <show config:type="boolean">true</show>
+    <timeout config:type="integer">10</timeout>
+    <log config:type="boolean">true</log>
+  </yesno_messages>
+</report>
+```
+
+## The New AutoYaST Issues Mechanism
+
+In openSUSE Leap 15.0 and SUSE Linux Enterprise SLE 15, AutoYaST introduced a mechanism to report
+semantic issues in the profile. At the beginning, it was implemented as part of the storage-ng
+initiative, and later it was [generalized](https://github.com/yast/yast-autoinstallation/pull/431)
+to be used by other parts of AutoYaST. It is still a work in progress, so expect small changes in
+the API.
+
+The core of the implementation lives in the [Installation::AutoinstIssues
+module](https://github.com/yast/yast-yast2/blob/5902d449f108bba6edcbccad8394ed93dc9cae39/library/general/src/lib/installation/autoinst_issues).
+Each kind of issue is represented by a class which inherits from [Installation::AutoinstIssues::Issue
+class](https://github.com/yast/yast-yast2/blob/5902d449f108bba6edcbccad8394ed93dc9cae39/library/general/src/lib/installation/autoinst_issues/issue.rb).
+Classes for common problems, like `InvalidValue` and `MissingValue`, are already offered but you are
+free to implement your own. For instance, see the
+[Y2Storage::AutoinstIssues](https://github.com/yast/yast-storage-ng/tree/master/src/lib/y2storage/autoinst_issues)
+for additional examples.
+
+All reported problems are added to an [IssuesList
+instance](https://github.com/yast/yast-yast2/blob/5902d449f108bba6edcbccad8394ed93dc9cae39/library/general/src/lib/installation/autoinst_issues/list.rb)
+which is accessible through the `Yast::AutoInstall` module. So, in order to register a problem, you
+use the `IssuesList#add` method.
+
+```ruby
+Yast::AutoInstall.issues_list.add(
+  ::Installation::AutoinstIssues::InvalidValue,
+  firewall_section, # the firewall section is an instance of a Installation::AutoinstProfile::SectionWithAttributes subclass
+  "FW_DEV_INT",
+  "1",
+  _("It is not supported anymore."))
+)
+```
+
+An important difference with existing error reporting mechanism in (Auto)YaST is that the messages
```

Comment:
I am not sure if both mechanism serve same purpose. For me it looks like AutoinstIssues should be used exclusively for autoinst problems...so import, export and maybe catching exceptions in auto client and translate it to issue. On other hand Report can be used everywhere, even in CLI and it just put correct output. So it should be used in shared parts of code that is not autoyast exclusive.

Reply:
No, they do not serve to the same purpose. I will rephrase it.